Autodesk/synthesis
# Multiplayer robot selection

~~Blocked by #869.~~

### Description

Implement client robot selection during connection to multiplayer lobby. Also added a basic outline for what the server and client modes are going to look like.

Not my best code and I would be in favor of re-writing it later but it's a good outline for what multiplayer should look like.

### Objectives
- [x] Create test mode for starting an empty server.
- [x] Create test mode for connecting to an empty server.
- [x] Implement connected client downloading server side robots.
- [x] Create menu for robot selection.
- [x] Move away from testing modes.

### To Test

- Build the project and run it.
- Open up a server in either the unity editor or the built application.
- Then connect to it in either the unity editor or the built application.
- Send some robot data and pick a robot.
- Check out the list of connected clients.

[JIRA Issue - Client Upload Robot Data](https://jira.autodesk.com/browse/AARD-1561)
[JIRA Issue - Multiplayer Robot Selection](https://jira.autodesk.com/browse/AARD-1516)
